rpcserverhandlers_test.go

```diff
@@ -236,9 +244,15 @@ func (c *testRPCChain) CountVoteVersion(version uint32) (uint32, error) {
 	return c.countVoteVersion, nil
 }
 
-// EstimateNextStakeDifficulty a mocked estimated next stake difficulty.
+// EstimateNextStakeDifficulty returns a mocked estimated next stake difficulty.
+// The return values are stored as a queue so that it can be called multiple
+// times with different results.
 func (c *testRPCChain) EstimateNextStakeDifficulty(newTickets int64, useMaxTickets bool) (int64, error) {
-	return c.estimateNextStakeDifficulty, nil
+	diff, err := c.estimateNextStakeDifficultyQueue[0].diff, c.estimateNextStakeDifficultyQueue[0].err
+	if len(c.estimateNextStakeDifficultyQueue) > 1 {
+		c.estimateNextStakeDifficultyQueue = c.estimateNextStakeDifficultyQueue[1:]
+	}
+	return diff, err
```

Comment:
Yes that works too, but not if the method takes no parameters. This one does of course. But then mid/user returns would just be something random I guess, like `case newTickets == 3`? Although, FIFO has problems too, like requiring quite some thought when creating, as the order is important. The test can easily be broken by changing up the handler. Leaving for now, but no problem changing.

Reply:
Since this is just a function, the test case could implement pretty much anything (return based on params as above, just return something hardcoded, or use a queue or other data structure such as your example). Sounds good to leave for now.

Reply:
Oh, I see.

Reply:
Ok! You're right this is perfect. Sorry your explanation flew over my head the first time. Thanks for not giving up on me!